.github/workflows/test.yml

```diff
@@ -0,0 +1,112 @@
+# This is a GitHub workflow defining a set of jobs with a set of steps.
+# ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
+#
+name: Tests
+
+on:
+  pull_request:
+  push:
```

Comment:
I like having `workflow_dispatch:` here as well, it provides maintainers with a button to trigger the workflow manually. This can be useful to verify how tests work on current master for example. It is relevant whenever you observe a failure in a PR and get confused if it is because of the PR or the master.